crates/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -118,8 +120,20 @@ impl<'a> FunctionContext<'a> {
         match literal {
             ast::Literal::Array(array) => {
                 let elements = vecmap(&array.contents, |element| self.codegen_expression(element));
-                let typ = Self::convert_non_tuple_type(&array.typ);
-                self.codegen_array(elements, typ)
+                let contents_typ = Self::convert_non_tuple_type(&array.typ);
+                
+                let typ = Self::convert_type(&array.typ);
+                if typ.count_leaves() > 1 {
```

Comment:
Lets get a comment here what we're actually checking for. The implication is `leaf_count > 1 => slice` but we should spell it out more for new contributors.

Actually, since we have `array.typ` let's just be more direct and use that. Then we wouldn't need a comment.